### PR TITLE
Add OAuth token-based authentication to guide

### PR DESCRIPTION
**Pull request by XuanWang-Amos**
Add guide for OAuth token-based authentication

### CODE REVIEW

File: content/en/docs/guides/auth.md [hunk 1]
```diff
@@ -112,6 +112,26 @@ passed to the factory method.
...
+OAuth 2.0 Protocol is the industry-standard protocol for authorization. It enables
+websites or applications to obtain limited access to user accounts using OAuth tokens.
+
+gRPC offers a set of simple APIs to integrate OAuth 2.0 into applications, streamlining authentication.
+
+At a high level, using OAuth token-based authentication includes 3 steps:
+
+1. Get OAuth token on client side.
+   * You can obtain OAuth tokens using libraries like
+   [oauth2 package](https://pkg.go.dev/golang.org/x/oauth2#section-readme) in Go and
```
Comment (matthewstevenson88) [inline content/en/docs/guides/auth.md:126]: IIUC, this library is primarily for sending an existing OAuth tokens around, refreshing them, etc and it doesn't deal with generating/obtaining OAuth tokens (except for the subpackage https://pkg.go.dev/golang.org/x/oauth2/google, which is Google-specific).

Similarly the `google-auth` link below is Google-specific.

Since this top section is generic and not Google-specific, should we just assume that the user already has an existing OAuth token? WDYT?

File: content/en/docs/guides/auth.md [hunk 2]
```diff
@@ -112,6 +112,26 @@ passed to the factory method.
...
+   [google-auth](https://google-auth.readthedocs.io/en/master/user-guide.html) in Python.
+2. Create Channel credentials or per-call credentials with the OAuth token.
+   * The token will be send to server, normally as part of metadata.
```
Comment (matthewstevenson88) [inline content/en/docs/guides/auth.md:129]: If you want, you can replace "metadata" with the "HTTP Authorization header".

File: content/en/docs/guides/auth.md [hunk 3]
```diff
@@ -112,6 +112,26 @@ passed to the factory method.
...
+
+gRPC offers a set of simple APIs to integrate OAuth 2.0 into applications, streamlining authentication.
+
+At a high level, using OAuth token-based authentication includes 3 steps:
+
+1. Get OAuth token on client side.
+   * You can obtain OAuth tokens using libraries like
+   [oauth2 package](https://pkg.go.dev/golang.org/x/oauth2#section-readme) in Go and
+   [google-auth](https://google-auth.readthedocs.io/en/master/user-guide.html) in Python.
+2. Create Channel credentials or per-call credentials with the OAuth token.
```
Comment (matthewstevenson88) [inline content/en/docs/guides/auth.md:128]: nit: I think the OAuth token is always part of the per-call credentials, and then you might attach the per-call credentials to some channel credentials.

File: content/en/docs/guides/auth.md [hunk 4]
```diff
@@ -201,6 +221,27 @@ and authorization in various languages.
...
+| Language | Example                                   | Documentation                 |
+|----------|-------------------------------------------|-------------------------------|
+| C++      | N/A                                       | N/A                           |
+| Go       | [Go OAuth Example]                        | [Go OAuth Documentation]      |
+| Java     | [Java OAuth Example]                      | [Java OAuth Documentation]    |
+| Python   | [Python OAuth Example]                    | [Python OAuth Documentation]  |
+
+
+[Go OAuth Example]: https://github.com/grpc/grpc-go/tree/master/examples/features/authentication#authentication
+[Go OAuth Documentation]: https://github.com/grpc/grpc-java/tree/master/examples/example-oauth#authentication-example
```
Comment (matthewstevenson88) [inline content/en/docs/guides/auth.md:238]: This link points to Java, not Go.

**Review by matthewstevenson88 — COMMENTED** [review]
Thanks so much for adding this documentation! Left a few minor comments. :)

File: content/en/docs/guides/auth.md (hunk 1 above)
Comment (XuanWang-Amos) [reply content/en/docs/guides/auth.md:126]: I see, I think that make sense, let me rephrase this part.

File: content/en/docs/guides/auth.md (hunk 3 above)
Comment (XuanWang-Amos) [reply content/en/docs/guides/auth.md:128]: Got it, I'll add this.

File: content/en/docs/guides/auth.md (hunk 2 above)
Comment (XuanWang-Amos) [reply content/en/docs/guides/auth.md:129]: Make sense, I'll use header instead since it's more accurate.

File: content/en/docs/guides/auth.md (hunk 4 above)
Comment (XuanWang-Amos) [reply content/en/docs/guides/auth.md:238]: Nice catch!

**Review by XuanWang-Amos — COMMENTED** [review]
Thanks for reviewing!

**Review by matthewstevenson88 — APPROVED** [review]
Looks great! Thanks again for making these changes.

Please also get an approval from one of the leads, since I don't believe I have write access for grpc.io. :)

File: content/en/docs/guides/auth.md [hunk 5]
```diff
@@ -112,6 +112,26 @@ passed to the factory method.
...
+gRPC offers a set of simple APIs to integrate OAuth 2.0 into applications, streamlining authentication.
+
+At a high level, using OAuth token-based authentication includes 3 steps:
+
+1. Get or generate an OAuth token on client side.
+   * You can generate Google-specific tokens following instructions below.
+2. Create credentials with the OAuth token.
+   * OAuth token is always part of per-call credentials, you can also attach the per-call credentials
+   to some channel credentials.
+   * The token will be send to server, normally as part of HTTP Authorization header.
```
Comment (temawi) [inline content/en/docs/guides/auth.md:129]: "will be sent"

**Review by temawi — APPROVED** [review]
Just one typo to fix.

**Comment by temawi** [comment]
Thanks Xuan, I'll merge it.

**Comment by Paullevey** [comment]
No  how can my acount stay 0    its not.  Corectt

**Comment by Paullevey** [comment]
All assets is belangs to to me NOT  x  (VERONICA BESTER) SHE KEEP HANKEDING ME
